From 249e6b28c7840041e31db2721a1e733649c382e1 Mon Sep 17 00:00:00 2001
From: Locietta <locietta@qq.com>
Date: Wed, 19 Jun 2024 13:29:17 +0800
Subject: [PATCH] nf_nat_fullcone: fix clang uninitialized var werror

Do NOT use `unreachabe()`, which will produce empty basic blocks, causing objtool "can't
find jump dest instruction". We could use `unreachable()` and add `-mllvm -trap-unreachable` to build flags,
but just giving a default value is a much more straight-forward and clean solution here.

Signed-off-by: Locietta <locietta@gmail.com>
---
 net/netfilter/nf_nat_fullcone.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/netfilter/nf_nat_fullcone.c b/net/netfilter/nf_nat_fullcone.c
index c63470fd8..40ddb3b52 100644
--- a/net/netfilter/nf_nat_fullcone.c
+++ b/net/netfilter/nf_nat_fullcone.c
@@ -1379,6 +1379,8 @@ static unsigned int nf_nat_handle_postrouting(u8 nfproto, struct sk_buff *skb, u
 			is_src_mapping_active = src_mapping != NULL && check_mapping(src_mapping, net, zone);
 		} else if (nfproto == NFPROTO_IPV6) {
 			is_src_mapping_active = src_mapping_6 != NULL && check_mapping6(src_mapping_6, net, zone);
+		} else {
+			is_src_mapping_active = false;
 		}
 
 		if (is_src_mapping_active) {
@@ -1430,6 +1432,8 @@ static unsigned int nf_nat_handle_postrouting(u8 nfproto, struct sk_buff *skb, u
 
 				want_port =
 				    find_appropriate_port6(net, zone, original_port, &newrange->min_addr, range);
+			} else {
+				want_port = 0;
 			}
 
 			newrange->flags = NF_NAT_RANGE_MAP_IPS | NF_NAT_RANGE_PROTO_SPECIFIED;
-- 
2.45.1

